### PR TITLE
Add more item search / add options, and access group search

### PR DESCRIPTION
**Pull request by grahamc**

### CODE REVIEW

File: security-framework/src/item.rs [hunk 1]
```diff
@@ -592,11 +633,30 @@ impl ItemAddOptions {
                 },
             }
         }
-
+        let account_name = self.account_name.as_deref().map(CFString::from);
```
Comment (kornelski) [inline security-framework/src/item.rs:636]: I've just noticed that this and previous code are vulnerable to use after free. 

to_void gives a pointer to a temporary copy made by `CFString::from`. 

It should be giving a pointer to original longer lived CFString stored in self, or remove it from self and pass ownership, preventing Rust from decrementing refcount.

**Review by kornelski — CHANGES_REQUESTED** [review]

File: security-framework/src/item.rs (hunk 1 above)
Comment (grahamc) [reply security-framework/src/item.rs:636]: Ok! so more like `pub struct ItemSearchOptions {`, which stores the CFString directly in the struct? Can do.

Comment (grahamc) [reply security-framework/src/item.rs:636]: Hey so unfortunately I'm not real confident in my ability to do this correctly. I'm not real familiar with CF or unsafe Rust. I took a look into this, but never felt good enough about what I was trying.

Comment (kornelski) [reply security-framework/src/item.rs:636]: Sorry, false alarm! I've checked the implementation of `dict.add`, and it increases the reference counter. So the combo of {new string, give pointer, drop string} is valid.

Comment (grahamc) [reply security-framework/src/item.rs:636]: 🎉  thank you!

**Pull request merged**